### DISCUSSION TITLE
Feature debounce for react-instantsearch

### DISCUSSION THREAD

**Discussion by kopax-polyconseil** [algolia/instantsearch#5271] [General]
Hello,

I would like to debounce my input instant search instead of handling this debounce userland. 

At first, it seemed logical to me that it must be implemented it in the module, but after reading your documentation here:  https://www.algolia.com/doc/guides/building-search-ui/going-further/optimize-search-request-usage/react-hooks/ it appears that you are aware that too many queries are sent on each keystroke, but you don't propose a debounce for it.

Why this module wasn't designed at first with a debounce feature, and will you support this feature soon or later ?

Thanks :)

**Comment by Haroenv** [comment]
There's a guide here: https://www.algolia.com/doc/guides/building-search-ui/going-further/improve-performance/react-hooks/#disabling-as-you-type with code available here https://github.com/algolia/doc-code-samples/tree/master/react-instantsearch-hooks/conditional-debouncing ([live](https://githubbox.com/algolia/doc-code-samples/tree/master/react-instantsearch-hooks/conditional-debouncing)

**Reply by kopax-polyconseil** [reply]
Hi and thanks for your reply and for the links, that I have already been through.

My question was not a how to, but why is this not included as a feature ? 
It's a common need and you don't need anything from the app to maintain and propose it out of the box. 
Will it soon or later be supported ? I think it should not be handled user lands.

Also, your snippet is similar to a throttling, not a debouncing.
